Sources/Sentry/SentrySysctl.m

```diff
@@ -1,5 +1,17 @@
 #import "SentrySysctl.h"
 #import "SentryCrashSysCtl.h"
+#include <stdio.h>
+#include <time.h>
+
+static time_t mod_init_time;
+
+void
+sentry_mod_init_hook(int argc, char **argv, char **envp)
+{
+    time(&mod_init_time);
+}
+__attribute__((section("__DATA,__mod_init_func"))) typeof(sentry_mod_init_hook) *__init
+    = sentry_mod_init_hook;
```

Comment:
@armcknight, could you please have a look at this c code?

Reply:
https://github.com/getsentry/sentry-native/blob/f6eda12d7180e3801dc3ac9def8c3c61048137c4/src/sentry_utils.h#L99-L122

That should be it if you want unix timestamps. However that is not a guaranteed monotonic clock (it can go backwards depending on multithreading and OS scheduling)

But I think depending on your usecase non-monotonic time should be fine.

Reply:
Thanks, @Swatinem.

Reply:
[SentryCrashReportStore](https://github.com/getsentry/sentry-cocoa/blob/master/Sources/SentryCrash/Recording/SentryCrashReportStore.c#L163-L178) also has a sample